File: backend/router/tools/code_generator.go
package tools

import (
	"k-admin-system/api/v1/tools"
	"k-admin-system/global"
	"k-admin-system/middleware"
	toolsService "k-admin-system/service/tools"

	"github.com/gin-gonic/gin"
)

// InitCodeGeneratorRouter 初始化代码生成器路由
func InitCodeGeneratorRouter(router *gin.RouterGroup) {
	service := toolsService.NewCodeGeneratorService(global.DB)
	codeGenApi := &tools.CodeGeneratorAPI{
		Service: service,
	}

	// 所有Code Generator路由都需要JWT认证和管理员权限
	genGroup := router.Group("/gen")
	genGroup.Use(middleware.JWTAuth())
	// TODO: 添加Casbin中间件检查管理员权限
	// genGroup.Use(middleware.CasbinAuth())
	{
		// 获取表元数据
		genGroup.GET("/metadata/:tableName", codeGenApi.GetTableMetadata)

		// 代码生成
		genGroup.POST("/preview", codeGenApi.PreviewCode)
		genGroup.POST("/generate", codeGenApi.GenerateCode)

		// 表创建
		genGroup.POST("/table", codeGenApi.CreateTable)
	}
}

File: frontend/src/api/codeGenerator.ts
import request from '../utils/request';

/**
 * Code Generator API definitions
 */

export interface FieldConfig {
  columnName: string;
  fieldName: string;
  fieldType: string;
  jsonTag: string;
  gormTag: string;
  comment: string;
}

export interface TableMetadata {
  tableName: string;
  fields: FieldConfig[];
}

export interface GenerateOptions {
  generateModel: boolean;
  generateService: boolean;
  generateAPI: boolean;
  generateRouter: boolean;
  generateFrontendAPI: boolean;
  generateFrontendTypes: boolean;
  generateFrontendPage: boolean;
}

export interface GenerateConfig {
  tableName: string;
  structName: string;
  packageName: string;
  frontendPath: string;
  fields: FieldConfig[];
  options: GenerateOptions;
}

export interface GeneratedFiles {
  [filePath: string]: string; // file path -> file content
}

// Get table metadata
export const getTableMetadata = (tableName: string): Promise<TableMetadata> => {
  return request.get(`/api/v1/tools/gen/metadata/${tableName}`);
};

// Preview generated code
export const previewCode = (config: GenerateConfig): Promise<GeneratedFiles> => {
  return request.post('/api/v1/tools/gen/preview', config);
};

// Generate code and write to files
export const generateCode = (config: GenerateConfig): Promise<GeneratedFiles> => {
  return request.post('/api/v1/tools/gen/generate', config);
};

// Create table from field definitions
export interface CreateTableRequest {
  tableName: string;
  fields: FieldConfig[];
}

export const createTable = (data: CreateTableRequest): Promise<void> => {
  return request.post('/api/v1/tools/gen/create-table', data);
};
